src/main/java/org/gaul/s3proxy/junit/S3ProxyJunitCore.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2014-2020 Andrew Gaul <andrew@gaul.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.s3proxy.junit;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.Properties;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.gaul.s3proxy.AuthenticationType;
+import org.gaul.s3proxy.S3Proxy;
+import org.jclouds.ContextBuilder;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.BlobStoreContext;
+import org.jclouds.blobstore.domain.StorageMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3ProxyJunitCore {
+
+    private static final Logger logger = LoggerFactory.getLogger(
+            S3ProxyExtension.class);
+
+    private static final String LOCALHOST = "127.0.0.1";
+
+    private final String accessKey;
+    private final String secretKey;
+    private final String endpointFormat;
+    private final S3Proxy s3Proxy;
+
+    private final BlobStoreContext blobStoreContext;
+    private URI endpointUri;
+    private final File blobStoreLocation;
+
+    public static final class Builder {
+        private AuthenticationType authType = AuthenticationType.NONE;
+        private String accessKey;
+        private String secretKey;
+        private String secretStorePath;
+        private String secretStorePassword;
+        private int port = -1;
+        private boolean ignoreUnknownHeaders;
+        private String blobStoreProvider = "filesystem";
+
+        Builder() { }
```

Comment:
I think you can remove this constructor, the compiler will generate one, not sure we need a default one?